sdk/python/kfp/dsl/component_decorator.py

```diff
@@ -116,7 +120,8 @@ def pipeline():
             output_component_file=output_component_file,
             install_kfp_package=install_kfp_package,
             kfp_package_path=kfp_package_path,
-            pip_trusted_hosts=pip_trusted_hosts)
+            pip_trusted_hosts=pip_trusted_hosts,
+            use_venv=use_venv)
```

Comment:
Would be nice to have a unit test for this `use_venv` functionality, including an edge case where the environment might have read-only restrictions.

Reply:
I can work on that. Can you recommend the right place for both unit test (`sdk/python/kfp/dsl/component_decorator_test.py`?) and the edge case? Thanks!

Reply:
the tests that are included seem sufficient to me, is there anything not covered there that you are looking to address @DharmitD ?

Reply:
I was thinking of an edge case where the venv set up fails, but I agree, existing tests seem sufficient.